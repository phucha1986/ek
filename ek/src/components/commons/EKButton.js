import React, { useState } from 'react';

const EKButton = (params) => {
  const [isLoading, setIsLoading] = useState(false);
  const onHandleSubmit = async () =>
  { 
    if(typeof params.clientSideValidation !== "function" || !params.clientSideValidation() || isLoading)
    {
      return;
    }  

    if(params.EnableLoading)
    {           
      setIsLoading(true);         
      let handleResult = await params.handleSubmit();
      if(handleResult == params.SuccessCode)
      {
        setIsLoading(false);
      }

    }else{
      params.handleSubmit();
    }
  };

  return (
    <div className="login-form__action-btn">
      <button id={params.id} aria-disabled="false" className="call-to-action call-to-action__primary call-to-action--middle" onClick={onHandleSubmit}>
        {isLoading && 
          <div className="login-form__button-content">
            <div role="region" aria-label="The information is loading" className="ek-spinner ek-spinner__xsmall login-form__button-loader"></div>
            <p className="login-form__loading-label" aria-live="off">{params.LoadingText}</p>
          </div>
        }
        {!isLoading && params.title}
      </button>

      
    </div>   
  );
}

export default EKButton;
