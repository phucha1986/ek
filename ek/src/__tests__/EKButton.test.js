import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EKButton from '../components/commons/EKButton';

describe('EKButton', () => {
  it('renders without prop', () => {
    const wrapper = shallow(<EKButton />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('renders with prop', () => {
    const props = {
      id: 'btnTest',
      title: 'click',
      LoadingText: 'loading',
      clientSideValidation: jest.fn(),
      handleSubmit: jest.fn(),
      dispatch: jest.fn()      
    }
    const wrapper = shallow(<EKButton {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('should call the onclick when button is clicked', () => {
    const handleSubmit = jest.fn();
    const clientSideValidation = jest.fn();
    const props = {
      id: 'btnTest',
      title: 'click',
      LoadingText: 'loading',
      EnableLoading: false,
      clientSideValidation,
      handleSubmit,
      dispatch: jest.fn()      
    }
    const component = shallow(<EKButton {...props}/>);  
    component.find('button#btnTest').simulate('click');
    expect(clientSideValidation).toHaveBeenCalled();    
  });
  it('should return false if clientSideValidation is failed', () => {
    const handleSubmit = jest.fn();
    const clientSideValidation = jest.fn(() =>  false);
    const props = {
      id: 'btnTest',
      title: 'click',
      LoadingText: 'loading',
      EnableLoading: false,
      clientSideValidation,
      handleSubmit,
      dispatch: jest.fn()      
    }
    const component = shallow(<EKButton {...props}/>);  
    component.find('button#btnTest').simulate('click');
    expect(clientSideValidation.mock.results[0].value).toBeFalsy();    
  });
  it('should call handleSubmit if clientSideValidation is passed', () => {
    const handleSubmit = jest.fn();
    const clientSideValidation = jest.fn(() =>  true);
    const props = {
      id: 'btnTest',
      title: 'click',
      LoadingText: 'test loading',
      EnableLoading: true,
      clientSideValidation,
      handleSubmit,
      dispatch: jest.fn()      
    }
    const component = shallow(<EKButton {...props}/>);  
    component.find('button#btnTest').simulate('click');
    expect(handleSubmit).toHaveBeenCalled();    
  });
})