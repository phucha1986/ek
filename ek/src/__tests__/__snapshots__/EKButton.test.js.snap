// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EKButton renders with prop 1`] = `
<div
  className="login-form__action-btn"
>
  <button
    aria-disabled="false"
    className="call-to-action call-to-action__primary call-to-action--middle"
    id="btnTest"
    onClick={[Function]}
  >
    click
  </button>
</div>
`;

exports[`EKButton renders without prop 1`] = `
<div
  className="login-form__action-btn"
>
  <button
    aria-disabled="false"
    className="call-to-action call-to-action__primary call-to-action--middle"
    onClick={[Function]}
  />
</div>
`;
